implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migration_result.rs

```diff
@@ -0,0 +1,70 @@
+use core::fmt::{Display, Formatter};
+use serde::Serialize;
+
+/// This enum models the result of executing one migration.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MigrationResult {
+    /// The migration was successful
+    MigrationSuccess,
+    /// The migration had a failure
+    SomeMigrationError(MigrationFailure),
+}
```

Comment:
Why not write `MigrationFailure` as an `Error` and use `Result<T, MigrationFailure>`? I can see it's easy to use it this way in other parts of the code, instead of doing the error conversion stuff, but it still feels a little bit weird.

Also, I'd remove the `Some` prefix in `SomeMigrationError` and rename `MigrationFailure` to `MigrationError` if possible.

Reply:
I modeled it like that to represent expected "errors" as regular values while other errors can be more unexpected or transient, for example if we can't access the database. This is also why I chose to call those "failures" and not "errors".

I renamed `SomeMigrationError` to `MigrationFailure`. I actually did not realize that this would not create a clash with the `MigrationFailure` struct.